static/js/message_list_view.js

```diff
@@ -1445,4 +1445,76 @@ export class MessageListView {
             message_container.status_message = false;
         }
     }
+
+    update_sticky_recipient_headers() {
+        const rows_length = this._rows.size;
+        if (!rows_length) {
+            /* No headers are present */
+            return;
+        }
+        /* Intentionally remove sticky headers class here to make calculations simpler. */
+        $(".sticky_header").removeClass("sticky_header");
+        /* visible_top is navbar top position + height for us. */
+        const visible_top = $message_viewport.message_viewport_info().visible_top;
```

Comment:
Woah, do you know why this module was imported as ` $message_viewport`? Might be a good prep commit to not put a surprising `$` there.

Reply:
A mistake I made in #20907, because this module’s camel-cased `scrollTop` method made it look like a jQuery object. Feel free to fix.

Reply:
Okay, I will do it in a separate PR since that change seems independent of what we are doing here.

Reply:
oh! I see that it is only present in this file as `$message_viewport` then, I will add a prep commit.